Library/Homebrew/cli/named_args.rb

```diff
@@ -103,6 +103,34 @@ def to_formulae_to_casks(only: parent&.only_formula_or_cask, method: nil)
                                                 .map(&:freeze).freeze
       end
 
+      # Returns formulae and casks after validating that a tap is present for each of them.
+      def to_formulae_and_casks_with_taps
+        formulae_and_casks_with_taps, formulae_and_casks_without_taps =
+          to_formulae_and_casks.partition do |formula_or_cask|
+            case formula_or_cask
+            when Formula, Cask::Cask
+              formula_or_cask.tap&.installed?
+            else
+              # TODO: Refactor methods so that Sorbet can tell that this is unreachable.
+              raise ArgumentError, "Not a formula or cask: #{formula_or_cask.class}"
+            end
+          end
+
+        return formulae_and_casks_with_taps if formulae_and_casks_without_taps.blank?
+
+        types = []
+        types << "formulae" if formulae_and_casks_without_taps.any?(Formula)
+        types << "casks" if formulae_and_casks_without_taps.any?(Cask::Cask)
+
+        odie <<~ERROR
+          These #{types.join(" and ")} are not in any locally installed taps!
+
+          - #{formulae_and_casks_without_taps.sort_by(&:to_s).join("\n- ")}
```

Comment:
```suggestion
            #{formulae_and_casks_without_taps.sort_by(&:to_s).join("\n  ")}
```

We don't tend to use `-` here just to make piping a bit easier.

Reply:
The new error message looks like this.

```
Error: These formulae are not in any locally installed taps!

  gcc
  zstd

You may need to run `brew tap` to install additional taps.
```